### PR TITLE
ReadTheDocs migration foundation

### PR DESCRIPTION
**Pull request by rossbernet**
Added mkdocs.yml and some .md files to start organizing documentation move to ReadTheDocs

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+theme: readthedocs
+
+pages:
+- Home: 
+  - Welcome to GeoTrellis: index.md
+  - User Guide: user_guide.md
+  - High Performance Scala: HighPerformanceScala.md
+- Tutorials:
+  - Quick Start: quickstart.md
+  - Setup: setup.md
```
Comment (fosskers) [inline mkdocs.yml]: Setup should go before Quick Start.

File: mkdocs.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+site_name: GeoTrellis
+site_author: GeoTrellis contributers
+repo_url: https://github.com/rossbernet/geotrellis/tree/master/docs
```
Comment (fosskers) [inline mkdocs.yml]: We can switch this to the real repo URL, since we do all our debugging/prototyping locally with `mkdocs`.

File: mkdocs.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+copyright: Copyright (c) 2015-2016, GT contributors
```
Comment (fosskers) [inline mkdocs.yml]: `Copyright (c) 2015-2016, GT contributors` -> `Copyright (c) 2012-2016, Azavea`

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+(let us know what you're working on!) at:
+
+  - [Gitter](https://gitter.im/geotrellis/geotrellis)
+  - [GeoTrellis mailing list](https://groups.google.com/group/geotrellis-user)
+
+
+
+
+## Hello Raster!
+
```
Comment (fosskers) [inline docs/index.md]: Let's revisit if we should keep this example in this file. Open question: do we want to be showing any code in the first thing they see when they visit our docs?

Comment (fosskers) [inline docs/index.md]: If we do keep it here, it would be handy to have a small explanation of what it is.

Comment (fosskers) [inline docs/index.md]: We've opted to keep it, as it fits the target audience of GeoTrellis.

**Comment by rossbernet** [comment]
@lossyrob or @echeipesh this is merge ready, and Colin and Jake are looking to build on it.

**Pull request merged**